sopel/irc/abstract_backends.py

```diff
@@ -67,16 +67,15 @@ def decode_line(self, line: bytes) -> str:
         # We can't trust clients to pass valid Unicode.
         try:
             data = str(line, encoding='utf-8')
-        except UnicodeDecodeError:
+        except UnicodeDecodeError as e:
+            # ...unless the server announces UTF8ONLY
+            if "UTF8ONLY" in self.isupport:
+                raise e
```

Comment:
Wow, this is the perfect place for `e.add_note("UTF8ONLY server sent a line that could not be decoded as UTF-8")`, but that requires Python 3.11 🙁 (or a nasty monkeypatch, I guess)

Reply:
Yeah. Not a bad place for a regular LOGGER.warning() though.

Reply:
Never mind, there's no LOGGER in there... meh